ethcore/src/engines/clique/mod.rs

```diff
@@ -713,6 +714,14 @@ impl Engine<EthereumMachine> for Clique {
 						header.set_difficulty(DIFF_NOTURN);
 					}
 				}
+
+				let zero_padding_len = VANITY_LENGTH - header.extra_data().len();
+				if zero_padding_len > 0 {
```

Comment:
Looks ok to me but this piece of code code be simplified by using `Vec::resize`

Something like 
```rust
let mut extra_data = header.extra_data().clone();
extra_data.resize(VANITY_LENGTH, 0);
header.set_extra_data(new_extra_data);
```